aten/src/ATen/native/Math.h

```diff
@@ -3054,6 +3054,14 @@ inline C10_HOST_DEVICE T hermite_polynomial_h_forward(T x, int64_t n) {
         return x + x;
     }
 
+    if (std::is_same_v<T, float> && n > 128) {
+        return std::numeric_limits<T>::quiet_NaN();
+    } else if (std::is_same_v<T, double> && n > 512) {
+        return std::numeric_limits<T>::quiet_NaN();
+    } else if (n > 1024) {
+        return std::numeric_limits<T>::quiet_NaN();
+    }
```

Comment:
All CUDA versions supports constexprs right? If so, can one define `template<typename T> constexpr getHermitianLimit() { return 1024};` with specializations for float and double?
Can we do something like
```suggestion
    if (n > getHermitianLimit<T>()) {
        return std::numeric_limits<T>::quiet_NaN();
    }
```